set.c

```diff
@@ -650,11 +650,13 @@ set_i_to_a(VALUE set)
  *  call-seq:
  *    to_set(klass = Set, *args, &block) -> self or new_set
  *
- *  Returns self if receiver is an instance of +Set+ and no arguments or
- *  block are given.  Otherwise, converts the set to another with
- *  <tt>klass.new(self, *args, &block)</tt>.
+ *  Without arguments, returns +self+ (for duck-typing in methods that
```

Comment:
My preference would be to change the call-seq to show no arguments, and only document the behavior with no arguments. We are removing argument handling right after the release of 4.0.